autofit/non_linear/result.py

```diff
@@ -201,12 +260,63 @@ def dict(self) -> dict:
         Human-readable dictionary representation of the results
         """
         return {
-            "max_log_likelihood": self.samples.max_log_likelihood_sample.model_dict(),
+            "max_log_likelihood": self.samples_summary.max_log_likelihood_sample.model_dict(),
+            "median pdf": self.samples_summary.median_pdf_sample.model_dict(),
         }
 
     @property
-    def samples(self):
-        return self._samples
+    def samples(self) -> Samples:
+        """
+        Returns the samples of the non-linear search, for example the MCMC chains or nested sampling samples.
+
+        When a model-fit is run the first time, the samples are passed into the result via memory and therefore
+        always available.
+
+        However, if a model-fit is resumed the samples are not available in memory and they only way to load them is
+        via the `samples.csv` file output on the hard-disk. This property handles the loading of the samples from
+        the `samples.csv` file if they are not available in memory.
+
+        Returns
+        -------
+        The samples of the non-linear search.
+        """
+
+        # TODO : This needs to load a samples class based on the samples type.
+
+        if self._samples is not None:
+            return self._samples

```

Comment:
Could use cached_property

Reply:
Not quite, this is because samples can be passed into `__init__` meaning we dont load via csv (e.g. to avoid I/O)

Reply:
Ah ofc